pgmpy/models/BayesianModel.py

```diff
@@ -1030,3 +1030,16 @@ def get_markov_blanket(self, node):
         blanket_nodes = set(blanket_nodes)
         blanket_nodes.remove(node)
         return list(blanket_nodes)
+    
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            network_struct1 = (self.nodes(), self.edges())
+            network_struct2 = (other.nodes(), other.edges())
+            cpd_self = [str(i) for i in self.cpds]
+            cpd_other = [str(i) for i in other.cpds]
```

Comment:
I don't understand what you are trying to do here. Could you please explain a bit ?

Reply:
A good way of comparing the CPDs could be to create a dict of the form `{node: node_cpd}` for both self and other and then compare these two. Since CPD already has an implementation of `__eq__`, the comparison should work fine.